src/Concordium/Client/GRPC2.hs

```diff
@@ -0,0 +1,2693 @@
+    , status :: !BakerConsensusInfoStatus
+    }
+
+instance FromProto Proto.NodeInfo where
+    type Output' Proto.NodeInfo = NodeInfo
+    fromProto nInfo = do
+        nDetails <- nInfo ^. ProtoFields.maybe'details
+        let details = case nDetails of
+                ProtoFields.NodeInfo'Bootstrapper _ -> Nothing
+                ProtoFields.NodeInfo'Node' node -> do
+                    n <- node ^. ProtoFields.maybe'consensusStatus
+                    case n of
+                        ProtoFields.NodeInfo'Node'NotRunning _ ->
+                            return NodeNotRunning
+                        ProtoFields.NodeInfo'Node'Passive _ ->
+                            return NodePassive
+                        ProtoFields.NodeInfo'Node'Active cInfo ->
+                            NodeActive <$> fromProto cInfo
+        peerVersion <- nInfo ^? ProtoFields.peerVersion
+        localTime <- fmap fst . fromProto =<< nInfo ^? ProtoFields.localTime
+        peerUptime <- fromProto =<< nInfo ^? ProtoFields.peerUptime
+        networkInfo <- fromProto =<< nInfo ^? ProtoFields.networkInfo
+        return NodeInfo{..}
+
+data NodeDetails = NodeNotRunning | NodePassive | NodeActive !BakerConsensusInfo
+
+instance FromProto Proto.NodeInfo'NetworkInfo where
+    type Output' Proto.NodeInfo'NetworkInfo = NetworkInfo
+    fromProto nInfo = do
+        nodeId <- nInfo ^? ProtoFields.nodeId . ProtoFields.value
+        peerTotalSent <- fromIntegral <$> nInfo ^? ProtoFields.peerTotalSent
+        peerTotalReceived <- fromIntegral <$> nInfo ^? ProtoFields.peerTotalReceived
+        avgBpsIn <- fromIntegral <$> nInfo ^? ProtoFields.avgBpsIn
+        avgBpsOut <- fromIntegral <$> nInfo ^? ProtoFields.avgBpsOut
+        return NetworkInfo{..}
+
+data NetworkInfo = NetworkInfo
+    { nodeId :: !Text
+    , peerTotalSent :: !Integer
+    , peerTotalReceived :: !Integer
+    , avgBpsIn :: !Integer
+    , avgBpsOut :: !Integer
+    }
+
+data NodeInfo = NodeInfo
+    { peerVersion :: !Text
+    , localTime :: !Timestamp
+    , peerUptime :: !Duration
+    , networkInfo :: !NetworkInfo
+    , details :: !(Maybe NodeDetails)
+    }
+
+instance FromProto Proto.ChainParametersV0 where
+    type Output' Proto.ChainParametersV0 = Parameters.ChainParameters' 'ChainParametersV0
+    fromProto cParams = do
+        _cpElectionDifficulty <- fromProto =<< cParams ^? ProtoFields.electionDifficulty
+        _cpExchangeRates <- do
+            euroPerEnergy <- fromProto =<< cParams ^? ProtoFields.euroPerEnergy
+            microCcdPerEuro <- fromProto =<< cParams ^? ProtoFields.microCcdPerEuro
+            return $ Parameters.makeExchangeRates euroPerEnergy microCcdPerEuro
+        _cpCooldownParameters <-
+            fmap Parameters.CooldownParametersV0
+                . fromProto
+                =<< cParams ^? ProtoFields.bakerCooldownEpochs
+        -- FIXME: I don't see this (required field) anywhere in the protobuf message.
+        _cpTimeParameters <- Nothing
```

Comment:
There are no time parameters for chain parameters 0.

So set `_cpTimeParameters = TimeParametersV0`.